plugins/flatpak/gs-flatpak.c

```diff
@@ -3678,6 +3679,76 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static guint64
+get_installation_dir_free_space (GsFlatpak *self, GError **error)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autoptr (GFileInfo) info = NULL;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+
+	info = g_file_query_filesystem_info (installation_dir,
+					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
+					     NULL,
+					     error);
+	if (info == NULL)
+		return 0;
+
+	return g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
+}
+
+static gboolean
+gs_flatpak_has_space_to_install (GsFlatpak *self, GsApp *app, GsAppList *list)
+{
+	g_autoptr(GError) error = NULL;
+	guint64 free_space = 0;
+	guint64 space_required = 0;
+
+	space_required = gs_app_get_size_download (app);
+	if (space_required == GS_APP_SIZE_UNKNOWABLE)
+		g_warning ("Failed to query download size: %s", gs_app_get_unique_id (app));
```

Comment:
GS_APP_SIZE_UNKNOWABLE is G_MAXUINT64 - it seems a bit weird to go ahead and add EOS_MIN_FREE_SPACE_NEEDED_BYTES to this - it will probably give us EOS_MIN_FREE_SPACE_NEEDED_BYTES - 1 but also causes an overflow condition. Would be better to explicitly set space_required to 0 if we can't fetch the value, wouldn't it?

Reply:
Oh! Great catch! I had the impresion that UNKNOWABLE was actually 0...
So yes, the `space_required` needs to be reset in that case.